File: 2nd_quarter/week02/Whaeun/덧칠하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, int m, vector<int> section) {
    int answer = 0;
    
    int start = section[0];
    int cnt = 0;
    
    for(int i = 0; i < section.size(); i++){
        
        if(section[i] <= start && i != 0){
            continue;
        }
        else{
            cnt++;
            start = section[i] + m - 1;
        }
    }
    
    answer = cnt;
    
    
    return answer;
}

File: 2nd_quarter/week02/Jose/덧칠하기.py

def solution(n, m, section):

    cnt = 0
    idx = 0 
    r = 0 
    
    # 깨달음 ! 시작을 0부터 할 필요가 없다는 사실을 
    while idx < len(section):
        # print(r, section[idx])
        if r < section[idx]:
            r = section[idx] + m - 1
            cnt +=1
        idx += 1 

    return cnt
